kolibri/plugins/user/assets/src/vue/sign-in-page/index.vue

```diff
@@ -1,16 +1,97 @@
 <template>
 
   <div>
-    <h1>Sign In</h1>
-    <p>Use this page for in-progress component prototyping.</p>
+    <img class="logo" alt="logo">
+    <h1>{{ $tr('signInInstantSchools') }}</h1>
+
+    <form ref="form" @submit.prevent="signIn">
+      <label for="username">{{ $tr('username') }}</label>
+      <input
+        id="username"
+        type="text"
+        :placeholder="$tr('enterUsername')"
+        :aria-label="$tr('username')"
+        v-model="username"
+        autocomplete="username"
+        required
+        autofocus>
+      <label for="password">{{ $tr('password') }}</label>
+      <input
+        id="password"
+        type="password"
+        :placeholder="$tr('enterPassword')"
+        :aria-label="$tr('password')"
+        v-model="password"
+        autocomplete="current-password"
+        required>
+      <icon-button :text="$tr('signIn')" :primary="true" type="submit"></icon-button>
+
+      <p v-if="loginError" class="sign-in-error">{{ $tr('signInError') }}</p>
+    </form>
+
+    <hr>
+
+    <p>{{ $tr('noAccount') }}</p>
+    <router-link :to="signUp">
+      <icon-button :text="$tr('createAccount')" :primary="true"></icon-button>
+    </router-link>
+
+    <router-link :to="learn">
+      <icon-button :text="$tr('accessAsGuest')" :primary="false"></icon-button>
+    </router-link>
   </div>
 
 </template>
 
 
 <script>
 
+  const actions = require('kolibri.coreVue.vuex.actions');
+
   module.exports = {
+    $trNameSpace: 'signInPage',
+    $trs: {
+      signInInstantSchools: 'Sign In to Instant Schools',
+      username: 'Username',
+      enterUsername: 'Enter Username',
+      password: 'Password',
+      enterPassword: 'Enter Password',
+      signIn: 'Sign In',
+      noAccount: 'Don\'t have an account?',
+      createAccount: 'Create Account',
+      accessAsGuest: 'Access as Guest',
+      signInError: 'Incorrect username or password',
+    },
+    data: () => ({
+      username: '',
+      password: '',
+    }),
+    computed: {
+      signUp() {
+        return '/signup';
+      },
+      learn() {
+        return '/learn';
+      },
+    },
+    methods: {
+      signIn() {
+        if (this.$refs.form.checkValidity()) {
```

Comment:
cool - hadn't seen this feature before. Is it considered best practice these days to embed as much of the validation logic in the HTML as possible?

(It's certainly more declarative/semantic; however I also wonder if it will scale to more sophisticated validation and interaction with components and JS logic)